Packs/jamf/Integrations/jamfV2/jamfV2.py

```diff
@@ -72,31 +81,62 @@ def _get_token(self) -> str:
 
     def _generate_token(self) -> str:
         """
-        Generates new token.
+        Generates new token and updates the integration context.
         """
-        resp = self._http_request(method='POST', url_suffix='api/v1/auth/token', resp_type='json')
-        token = resp.get('token')
-        expiration_time = int(dateparser.parse(resp.get('expires')).timestamp())  # type: ignore[union-attr]
+        if self.username and self.password:  # basic auth is used to create a token
+            token, expiration_time = self.generate_basic_auth_token()
+
+        elif self.client_id and self.client_secret:  # client id and client secret are used to create a token
+            token, expiration_time = self.generate_client_credentials_token()
+        else:
+            raise ValueError(
+                "Invalid authentication configuration."
+                "Please provide either username and password or client id and client secret.")
+
         integration_context = get_integration_context()
         integration_context.update({'token': token})
-        # Add 10 minutes buffer for the token
+        # subtract 60 seconds from the expiration time to make sure the token is still valid
         integration_context.update({'expires': expiration_time - 60})
         set_integration_context(integration_context)
 
         return token
 
+    def generate_basic_auth_token(self) -> tuple[str, int]:
+        """
+        Generates a token using basic authentication.
+        """
+        resp = self._http_request(method='POST', url_suffix='api/v1/auth/token', resp_type='json',
+                                  auth=(self.username, self.password))
+        token = resp.get('token')
+        expiration_time = int(dateparser.parse(resp.get('expires')).timestamp())    # type: ignore
+        return token, expiration_time
+
+    def generate_client_credentials_token(self) -> tuple[str, int]:
+        """
+        Generates a token using client credentials.
+        """
+        resp = self._http_request(method='POST', url_suffix="/api/v1/oauth/token", data={
+            'client_id': self.client_id,
+            'grant_type': 'client_credentials',
+            'client_secret': self.client_secret},
+            headers={"Content-Type": "application/x-www-form-urlencoded"}, resp_type='json')
+        token = resp.get('access_token')
+        now_timestamp = arg_to_datetime('now').timestamp()  # type:ignore
+        expiration_time = now_timestamp + resp.get('expires_in')
+        return token, expiration_time
+
     def _classic_api_post(self, url_suffix, data, error_handler):
         post_headers = ((self._headers or {}) | POST_HEADERS)  # merge the token and the POST headers
```

Comment:
```suggestion
       # If using a token, it will be available in the headers. Only when token creation failed, the basic auth is found in self.auth.
        post_headers = ((self._headers or {}) | POST_HEADERS)  # merge the token and the POST headers
```

Reply:
Done.